I've confirmed that this is Sphinx's fault, for adding a new `handler` to the `root` logging object.

If you pop an interactive `pdb` shell in the middle of e.g.:

```
$ PWNLIB_NOTERM=1 coverage run -m sphinx -b doctest docs/source docs/build/doctest pwnlib/util/web.py
```

```diff
diff --git a/pwnlib/tubes/remote.py b/pwnlib/tubes/remote.py
index 3673a006..6930e33b 100644
--- a/pwnlib/tubes/remote.py
+++ b/pwnlib/tubes/remote.py
@@ -85,6 +85,9 @@ class remote(sock):
         sock    = None
         timeout = self.timeout

+        import pdb
+        pdb.set_trace()
+
         with self.waitfor('Opening connection to %s on port %d' % (self.rhost, self.rport)) as h:
             for res in socket.getaddrinfo(self.rhost, self.rport, fam, typ, 0, socket.AI_PASSIVE):
                 self.family, self.type, self.proto, _canonname, sockaddr = res
```

We can inspect the logging stack.  Pwntool's default handler is not even installed yet, by virtue of the missing `[ERROR]` message.

```py
(Pdb) import pwnlib
(Pdb) pwnlib.context.context.log_level
40
(Pdb) self.error('asdf')
*** PwnlibException: asdf
(Pdb) pwnlib.log.install_default_handler()
(Pdb) self.error('asdf')
[ERROR] asdf
*** PwnlibException: asdf
```

Our filtering is working correctly, but Sphinx prints the stuff out:

```py
(Pdb) self.info('asdf')
asdf
(Pdb) pwnlib.context.context.log_level = 'info'
(Pdb) self.info('asdf')
[*] asdf
asdf
```

So Sphinx has installed some handler, and it's ignoring our log level information.  It has actually installed *three*.

```py
(Pdb) self._logger.root.handlers
[<sphinx.util.logging.NewLineStreamHandlerPY2 object at 0x104800250>, <logging.StreamHandler object at 0x104800550>, <sphinx.util.logging.MemoryHandler object at 0x105acf090>]
(Pdb) self._logger.root.handlers[1].stream
<sphinx.util.logging.LastMessagesWriter object at 0x104800510>
```

And with this information, we can easily find the source where this happens: https://github.com/sphinx-doc/sphinx/blob/228fdb892af25f4b93f2760f2cd6497f2aabc0be/sphinx/util/logging.py#L464-L493

```py
def setup(app, status, warning):
    # type: (Sphinx, IO, IO) -> None
    """Setup root logger for Sphinx"""
    logger = logging.getLogger()
    logger.setLevel(logging.NOTSET)

    # clear all handlers
    for handler in logger.handlers[:]:
        logger.removeHandler(handler)

    info_handler = NewLineStreamHandler(SafeEncodingWriter(status))  # type: ignore
    info_handler.addFilter(InfoFilter())
    info_handler.setLevel(VERBOSITY_MAP[app.verbosity])
    info_handler.setFormatter(ColorizeFormatter())
```

It seems that we should be able to override this function entirely in `conf.py` and kill this stupid shit by overriding the method.  However, that turns out to not be the case.  It looks like we can just manually remove the first `handler` object.

```
(Pdb) import logging
(Pdb) import pwnlib
(Pdb) pwnlib.log.install_default_handler()
(Pdb) pwnlib.context.context.log_level = 'info'
(Pdb) logging.root.handlers
[<sphinx.util.logging.NewLineStreamHandlerPY2 object at 0x10fad2250>, <logging.StreamHandler object at 0x10fad2550>, <sphinx.util.logging.MemoryHandler object at 0x110da1090>]
(Pdb) self.info('asdf')
[*] asdf
asdf
(Pdb) del logging.root.handlers[0]
(Pdb) self.info('asdf')
[*] asdf
```

--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -52,14 +52,15 @@ class SphinxPwnlibFilter(logging.Filter):
     def filter(self, record):
         if record.name.startswith('pwn'):
             return False
+        if record.name.startswith('paramiko'):
+            return False
         return True
 
 log_filter = SphinxPwnlibFilter()
 
 for i, handler in enumerate(logging.root.handlers):
-    if isinstance(handler, sphinx.util.logging.NewLineStreamHandler):
-        print("Filtering Sphinx handler", handler)
-        handler.addFilter(log_filter)
+    print("Filtering Sphinx handler", handler)
+    handler.addFilter(log_filter)
 
 # Napoleon settings
 napoleon_use_ivar = True

--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -44,6 +44,23 @@ extensions = [
     'sphinxcontrib.napoleon'
 ]
 
+# Disable "info" logging directly to stdout by Sphinx
+import logging
+import sphinx.util.logging
+
+class SphinxPwnlibFilter(logging.Filter):
+    def filter(self, record):
+        if record.name.startswith('pwn'):
+            return False
+        return True
+
+log_filter = SphinxPwnlibFilter()
+
+for i, handler in enumerate(logging.root.handlers):
+    if isinstance(handler, sphinx.util.logging.NewLineStreamHandler):
+        print("Filtering Sphinx handler", handler)
+        handler.addFilter(log_filter)
+
 # Napoleon settings
 napoleon_use_ivar = True
 napoleon_use_rtype = False

--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -46,7 +46,6 @@ extensions = [
 
 # Disable "info" logging directly to stdout by Sphinx
 import logging
-import sphinx.util.logging
 
 class SphinxPwnlibFilter(logging.Filter):
     def filter(self, record):

